### PR TITLE
BUG: RollingGroupby MultiIndex levels dropped

### LINKED ISSUE #38523: BUG: MultiIndex RollingGroupby returns only one level of index
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

```python
import pandas as pd
print(pd.__version__)

arrays = [['Falcon', 'Falcon', 'Parrot', 'Parrot'],
          ['Captive', 'Wild', 'Captive', 'Wild']]
index = pd.MultiIndex.from_arrays(arrays, names=('Animal', 'Type'))
df = pd.DataFrame({'Max Speed': [390., 350., 30., 20.]},
                  index=index)
print(df, '\n')
print(df.groupby(level=0)['Max Speed'].rolling(2).sum())
```

#### Problem description

Version 1.1.5 returned different result compared to earlier versions

#### Expected Output (0.22.0)

    0.22.0
                    Max Speed
    Animal Type              
    Falcon Captive      390.0
           Wild         350.0
    Parrot Captive       30.0
           Wild          20.0 
    
    Animal  Animal  Type   
    Falcon  Falcon  Captive      NaN
                    Wild       740.0
    Parrot  Parrot  Captive      NaN
                    Wild        50.0
    Name: Max Speed, dtype: float64

#### Expected Output (1.1.3)

    1.1.3
                    Max Speed
    Animal Type              
    Falcon Captive      390.0
           Wild         350.0
    Parrot Captive       30.0
           Wild          20.0 
    
    Animal  Animal  Type   
    Falcon  Falcon  Captive      NaN
                    Wild       740.0
    Parrot  Parrot  Captive      NaN
                    Wild        50.0
    Name: Max Speed, dtype: float64

#### Output (1.1.5)

    1.1.5
                    Max Speed
    Animal Type              
    Falcon Captive      390.0
           Wild         350.0
    Parrot Captive       30.0
           Wild          20.0 
    
    Animal
    Falcon      NaN
    Falcon    740.0
    Parrot      NaN
    Parrot     50.0
    Name: Max Speed, dtype: float64

<details>

INSTALLED VERSIONS
------------------
commit           : b5958ee1999e9aead1938c0bba2b674378807b3d
python           : 3.7.9.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 4, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.1.5
numpy            : 1.19.2
pytz             : 2020.4
dateutil         : 2.8.1
pip              : 20.3.3
setuptools       : 51.0.0.post20201207
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.2
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.19.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.15.1
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : None
tables           : None
tabulate         : 0.8.7
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### LINKED ISSUE #38787: API: Preferred MultiIndex result for groupby().rolling() for an object with a MultiIndex
`groupby().rolling()` in master currently constructs the resulting `MultiIndex` manually by inserting `groupby` keys as the first level(s) and then the original object's `Index` as the second level(s).

However, `groupby().rolling()` behaves similarly to `groupby().transform()` (i.e. maintains the original shape), so should the resulting index align with results of `groupby().transform()`?

```
In [4]: pd.__version__
Out[4]: '1.3.0.dev0+238.g1d196295c'

In [6]: df = pd.DataFrame({'a': [1], 'b': [2]})

In [7]: df
Out[7]:
   a  b
0  1  2

# Example of the original DataFrame having a regular Index
In [8]: df.groupby('a').rolling(1).sum()
Out[8]:
       a    b
a
1 0  1.0  2.0

In [16]: df.groupby('a').transform(lambda x: np.sum(x))
Out[16]:
   b
0  2

In [9]: df = pd.DataFrame({'a': [1], 'b': [2]}, index=pd.MultiIndex.from_tuples([('idx1', 'idx2')], names=['label1', 'label2']))

In [9]: df
Out[9]:
               a  b
label1 label2
idx1   idx2    1  2

# Examples of the original DataFrame having a MultiIndex
In [10]: df.groupby('label1').rolling(1).sum()
Out[10]:
                        a    b
label1 label1 label2
idx1   idx1   idx2    1.0  2.0

In [20]: df.groupby('label1').transform(lambda x: np.sum(x))
Out[20]:
               a  b
label1 label2
idx1   idx2    1  2

In [11]: df.groupby('a').rolling(1).sum()
Out[11]:
                   a    b
a label1 label2
1 idx1   idx2    1.0  2.0

In [21]: df.groupby('a').transform(lambda x: np.sum(x))
Out[21]:
               b
label1 label2
idx1   idx2    2

In [12]: df.groupby(['label1', 'a']).rolling(1).sum()
Out[12]:
                          b
label1 a label1 label2
idx1   1 idx1   idx2    2.0

In [22]: df.groupby(['label1', 'a']).transform(lambda x: np.sum(x))
Out[22]:
               b
label1 label2
idx1   idx2    2
```

As shown, when the original object as a `MultiIndex`, there is consistency of the resulting `MultiIndex` for the `groupby().rolling()` result but can lead to redundancy. There is lack of consistency of the resulting `MultiIndex` for the `groupby().transform()` result but looks more convenient.

IMO I prefer the consistent result we have today in `groupby().rolling()` but open to thoughts.

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #38523 
- [x] closes #38787
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Essentially https://github.com/pandas-dev/pandas/pull/38737 but targeted for 1.3 as it appears that the current behavior is largely unexpected

### CODE REVIEW

**Comment by jreback** [comment]
hmm should we target 1.2.4? e.g. how does this compare to that behavior (and how does that compare to 1.1.x)?

**Comment by mroeschke** [comment]
It looks like you mentioned before this would be okay for 1.3 and we reverted this before for 1.2.x

This gets us back to 1.1.3 behavior which was correct according to this bug report https://github.com/pandas-dev/pandas/issues/38523

**Comment by jreback** [comment]
> It looks like you mentioned before this would be okay for 1.3 and we reverted this before for 1.2.x
> 
> This gets us back to 1.1.3 behavior which was correct according to this bug report #38523

ok awesome

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -693,6 +693,7 @@ Groupby/resample/rolling
 - Bug in :class:`core.window.ewm.ExponentialMovingWindow` when calling ``__getitem__`` would not retain ``com``, ``span``, ``alpha`` or ``halflife`` attributes  (:issue:`40164`)
 - :class:`core.window.ewm.ExponentialMovingWindow` now raises a ``NotImplementedError`` when specifying ``times`` with ``adjust=False`` due to an incorrect calculation (:issue:`40098`)
 - Bug in :meth:`Series.asfreq` and :meth:`DataFrame.asfreq` dropping rows when the index is not sorted (:issue:`39805`)
+- Bug in :class:`core.window.rolling.RollingGroupby` where an level of a :class:`MultiIndex` would be dropped if it overlapped with a groupby label (:issue:`38787`, :issue:`38523`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:696]: let's make this a full sub-section and mention this goes back to 1.1.x behavior

Comment (mroeschke) [reply doc/source/whatsnew/v1.3.0.rst]: Done

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**